integration/test_collection.py

```diff
@@ -879,3 +880,44 @@ def test_empty_search_returns_everything(client: weaviate.Client):
     assert objects[0].metadata.creation_time_unix is not None
 
     client.collection.delete("TestReturnEverything")
+
+
+def test_insert_date_property(client: weaviate.Client):
+    collection = client.collection.create(
+        CollectionConfig(
+            name="TestInsertDateProperty",
+            vectorizer=Vectorizer.NONE,
+            properties=[Property(name="date", data_type=DataType.DATE)],
+        )
+    )
+
+    now = datetime.now(timezone.utc)
```

Comment:
The default behaviour of `datetime` in the absence of a user specified timezone is to not include one, which therefore violates the RFC3339 spec returning an error from Weaviate if done so. I'd say the middle road of assuming UTC in the absence of tzinfo is the most sensible. This way barrier to entry is reduced since devs need not worry about the `datetime` library's nuances but problems around local times are avoided.

Documentation will be required, however, for the edge case where devs batch load data without offsets that in fact required them. This is their issue of data quality but documentation should help guide them on this matter.